control-plane/csi-driver/src/bin/controller/controller.rs

```diff
@@ -1016,7 +1016,6 @@ impl rpc::csi::controller_server::Controller for CsiControllerSvc {
             .expand_volume(&vol_uuid, requested_size as u64)
             .await
             .map_err(|error| match error {
-                ApiClientError::PreconditionFailed(msg) => Status::failed_precondition(msg),
                 ApiClientError::ResourceExhausted(msg) => Status::out_of_range(msg),
```

Comment:
Also I wonder, do we need to remap resource exhausted to out of range?
Why not return resource exhausted status?

Reply:
I've done it because the spec.md lists this as a special case for invalid capacity_range. ResourceExhausted fits here as well. I've just concluded testing the retry durations for error cases. And all, but FailedPrecondition, are exponential backoff errors. The errors don't have anything different about then in the retry context, so we might as well go ahead pick the one which best fits the case.

Reply:
I'll change this to ResourceExhausted.